### PR TITLE
Enable ServiceMonitors per TempoStack instance

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
In the `.spec.observability.metrics.enabled` field of the TempoStack spec.

Resolves: #333

### CODE REVIEW

File: apis/tempo/v1alpha1/tempostack_types.go [hunk 1]
```diff
@@ -104,6 +111,16 @@ type ObservabilitySpec struct {
...
 }
 
+// MetricsConfigSpec defines a metrics config.
+type MetricsConfigSpec struct {
+	// Enabled specifies if a ServiceMonitor should be created for each Tempo component.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Enabled"
+	Enabled bool `json:"enabled,omitempty"`
```
Comment (pavolloffay) [inline apis/tempo/v1alpha1/tempostack_types.go:121]: I wonder if we should be more explicit in the name - e.g. `CreateServiceMonitor` . 

Also do we know if Tempo will stick with p8s metrics? Are there any plans to move to OTLP metrics? cc) @frzifus

File: apis/tempo/v1alpha1/tempostack_webhook_test.go [hunk 2]
```diff
@@ -877,18 +877,59 @@ func TestValidateTenantConfigs(t *testing.T) {
 }
 
 func TestValidatorObservabilityTracingConfig(t *testing.T) {
-	tracingBase := field.NewPath("spec").Child("template").Child("observability").Child("tracing")
+	observabilityBase := field.NewPath("spec").Child("observability")
+	metricsBase := observabilityBase.Child("tracing")
```
Comment (iblancasa) [inline apis/tempo/v1alpha1/tempostack_webhook_test.go:881]: Wouldn't it be?
```suggestion
	metricsBase := observabilityBase.Child("metrics")
```

File: apis/tempo/v1alpha1/tempostack_webhook.go [hunk 3]
```diff
@@ -267,7 +267,17 @@ func (v *validator) validateGateway(tempo TempoStack) field.ErrorList {
 }
 
 func (v *validator) validateObservability(tempo TempoStack) field.ErrorList {
-	tracingBase := field.NewPath("spec").Child("template").Child("observability").Child("tracing")
+	observabilityBase := field.NewPath("spec").Child("observability")
+	metricsBase := observabilityBase.Child("tracing")
```
Comment (iblancasa) [inline apis/tempo/v1alpha1/tempostack_webhook.go:271]: Maybe
```suggestion
	metricsBase := observabilityBase.Child("metrics")
```

File: apis/tempo/v1alpha1/tempostack_types.go (hunk 1 above)
Comment (frzifus) [reply apis/tempo/v1alpha1/tempostack_types.go:121]: y, i guess being a bit more explicit would make sense.

> Are there any plans to move to OTLP metrics?

Not that i would know of. I couldnt find a ticket and there is no otel metric instrumentation in their code.

Comment (andreasgerstmayr) [reply apis/tempo/v1alpha1/tempostack_types.go]: Ok, I've updated the field name and displayName.

**Review by frzifus — APPROVED** [review]

**Pull request merged**